pkg/machine/qemu/machine.go

```diff
@@ -236,7 +236,11 @@ func (q *QEMUStubber) Remove(mc *vmconfigs.MachineConfig) ([]string, func() erro
 	}
 
 	return qemuRmFiles, func() error {
-		return nil
+		if err := mc.QEMUHypervisor.QEMUPidPath.Delete(); err != nil {
+			return err
```

Comment:
if we fail here, we dont clean up the next one ... should above be a soft error (i.e. logrus) and the next one be the hard error (return err)